tokio-util/src/task/spawn_aborting.rs

```diff
@@ -0,0 +1,38 @@
+use std::ops::{Deref, DerefMut};
+
+use tokio::task::JoinHandle;
+
+use futures_core::Future;
+
+/// This is a wrapper type around JoinHandle that allows it to be dropped.
+#[derive(Debug)]
+pub struct DropHandle<T>(JoinHandle<T>);
+
+impl<T> Drop for DropHandle<T> {
+    fn drop(&mut self) {
+        self.0.abort();
+    }
+}
+
+impl<T> Deref for DropHandle<T> {
+    type Target = JoinHandle<T>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<T> DerefMut for DropHandle<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
```

Comment:
Instead of this, could you add all of the wrapper methods here? You can implement `AsRef<JoinHandle>` if you like.

It would probably also be nice with a `Future` implementation so that you can `.await` a `DropHandle`.